src/bytes.rs

```diff
@@ -265,19 +275,48 @@ impl<'a, T: Sized + AsRef<[u8]> + Send + Sync + 'a> From<Arc<T>> for Bytes<'a> {
     }
 }
 #[cfg(feature = "alloc")]
+impl From<alloc::boxed::Box<[u8]>> for Bytes<'_> {
+    fn from(value: alloc::boxed::Box<[u8]>) -> Self {
+        unsafe extern "C" fn release(this: *const (), capacity: usize) {
+            core::mem::drop(alloc::boxed::Box::from_raw(
+                core::ptr::slice_from_raw_parts_mut(this.cast::<u8>().cast_mut(), capacity),
+            ))
+        }
+        let data: &[u8] = value.as_ref();
+        Bytes {
+            start: core::ptr::NonNull::<[u8]>::from(data.as_ref()).cast(),
+            len: data.len(),
+            capacity: data.len(),
+            data: alloc::boxed::Box::into_raw(value) as *const (),
+            vtable: &BytesVt {
+                release: Some(release),
+                retain: None,
+            },
+        }
+    }
+}
+#[cfg(feature = "alloc")]
+impl From<alloc::vec::Vec<u8>> for Bytes<'_> {
+    fn from(value: alloc::vec::Vec<u8>) -> Self {
+        alloc::boxed::Box::<[u8]>::from(value).into()
+    }
+}
+#[cfg(feature = "alloc")]
 unsafe impl<'a> Send for Bytes<'a>
 where
     &'a [u8]: Send,
     Arc<[u8]>: Send,
+    alloc::boxed::Box<[u8]>: Sync,
```

Comment:
The good thing about the previous typo is that it has allowed me to spot this one 😆:

```suggestion
    alloc::boxed::Box<[u8]>: Send,
```